### ISSUE TITLE
关于生成伪标签代码配置与GMM原型数量选择 / About the configuration of pseudo-label generation code and the selection of the number of GMM prototypes

### ISSUE THREAD

**Issue by Kev111n** [cai525/Transformer4SED#10]
感谢您出色的开源工作！在研读论文和代码过程中，有两个问题需要向您咨询一下：

1. 生成伪标签的代码问题

我打算自己生成初始的GMM模型，查看项目后发现核心代码位于recipes\desed\pmam\gmm.py，但在生成伪标签时，我遇到下面的问题：

在recipes/desed/pmam/generate_pseudo_label.py中

- recipes/desed/pmam/setting.py 中缺失 get_configs() 函数定义
- --config_dir参数对应配置文件路径未知
- --feature_layer参数对应的网络层名未知

希望您能补齐初始GMM模型训练的相关代码和参数，以解答我的疑惑。

2. Prototype数量选择问题

我想应用自己的数据集到该模型，GMM的Prototype数量如何选择？原实验是30个，请问有什么技巧吗，是否与最终的分类数量10个有关？

期待您的专业见解，祝研究工作再创佳绩！

---

Thank you for your excellent open source work! In the process of studying the paper and code, I have two questions to ask you:

1. Code problem of generating pseudo labels

I plan to generate the initial GMM model by myself. After checking the project, I found that the core code is located in recipes\desed\pmam\gmm.py, but when generating pseudo labels, I encountered the following problems:

In recipes/desed/pmam/generate_pseudo_label.py

- The definition of the get_configs() function is missing in recipes/desed/pmam/setting.py
- The configuration file path corresponding to the --config_dir parameter is unknown
- The network layer name corresponding to the --feature_layer parameter is unknown

I hope you can complete the relevant code and parameters for the initial GMM model training to answer my doubts.

2. The problem of selecting the number of prototypes

I want to apply my own dataset to this model. How to choose the number of prototypes of GMM? The original experiment is 30. Is there any trick? Is it related to the final number of categories of 10?

Looking forward to your professional insights, I wish you more success in research!

**Comment by lzj1214** [comment]
不知道理解是否正确，训练流程分成几步
1. pre-train PaSST提取音频特征；
2. GMM初次建模并输出伪标签（frame-wise），伪标签用作训练encoder lora + Context Network；
3. 再进行GMM建模更新伪标签，再次训练encode lora + Context Network;
4. 固定encoder，针对特定任务进行半监督训练，固定encoder和context network。

如果有不对的地方有劳指出！如果可以的话，可以完善下readme的训练流程么？包括特征提取、gmm建模的命令行

**Comment by Kev111n** [comment]
Thank you for your detailed and patient reply, it is very helpful to me!😊
